src/screens/TopicPageScreen/index.js

```diff
@@ -413,31 +419,35 @@ const TopicPageScreen = (props) => {
         isFollow={isFollow}
         getTopicDetail={getTopicDetail}
       />
-      <TiktokScroll
-        ref={listRef}
-        contentHeight={dimen.size.TOPIC_CURRENT_ITEM_HEIGHT + normalizeFontSizeByWidth(4)}
-        data={feeds}
-        onEndReach={onEndReach}
-        onRefresh={onRefresh}
-        refreshing={loading}
-        renderItem={renderItem}
-        onScroll={handleScroll}
-        scrollEventThrottle={16}
-        showSearchBar={true}
-        searchHeight={isHeaderHide ? headerHideHeight : headerShowHeight}
-        snap
-        contentOffset={{x: 0, y: topPosition}}
-        contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
-        onMomentumScrollEnd={(event) => onWillSendViewPostTime(event, feeds)}
-      />
+
+      <View
+        style={{
+          marginTop: isHeaderHide ? -dimen.size.TOPIC_FEED_HEADER_HEIGHT : 0
+        }}>
+        <TiktokScroll
+          ref={listRef}
+          contentHeight={dimen.size.TOPIC_CURRENT_ITEM_HEIGHT + normalizeFontSizeByWidth(4)}
+          data={feeds}
+          onEndReach={onEndReach}
+          onRefresh={onRefresh}
+          refreshing={loading}
+          renderItem={renderItem}
+          onScroll={handleScroll}
+          scrollEventThrottle={16}
+          showSearchBar={true}
+          snap
+          contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
+          onMomentumScrollEnd={(event) => onWillSendViewPostTime(event, feeds)}
+        />
+      </View>
       <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} />
       <BlockComponent
         ref={refBlockComponent}
         refresh={onBlockCompleted}
         refreshAnonymous={onDeleteBlockedPostCompleted}
         screen="topic_screen"
       />
-    </SafeAreaProvider>
+    </View>
   );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `TiktokScroll` component now doesn't receive the `searchHeight` and `contentOffset` props. If these props were previously used to control the appearance and behavior of the component, their removal could potentially cause issues. The wrapping container of the entire return statement has been changed from `SafeAreaProvider` to `View`, which might affect the layout on devices with notches or rounded corners. 

Also, the `withInteractionsManaged` HOC has been removed from the export statement. If this HOC was providing some crucial functionality (like managing interactions), its removal could potentially cause issues.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
HOC makes it's slow, and not needed.
the searchHeigh and contentOffset it's also intended to be removed